### PR TITLE
docs initial commit

### PR DESCRIPTION
**Pull request by aprotopopov**

### CODE REVIEW

**Comment by CamDavidsonPilon** [comment]
Hey @aprotopopov - some of your previous commits are inside this PR. Looks like only the most recent commit is relevant.

**Comment by aprotopopov** [comment]
Yeah, you're right. Just verification, that structure is structure is ok. Now we need to define approximately what sections will be in the docs

**Comment by aprotopopov** [comment]
Moved **Quickstart** section from main README to separate file in docs/. What about divide Quickstart into subsections?

**Comment by aprotopopov** [comment]
Hmm.. Not sure why check failed for python 3.5 only .. May be there are some inconsistency in tests?

**Comment by CamDavidsonPilon** [comment]
I think it's just a flakey test. I restarted and it's 🍏 now

**Comment by aprotopopov** [comment]
Hey @CamDavidsonPilon, could you create current version of docs with the [readthedocs dashboard](https://readthedocs.org/accounts/login/?next=/dashboard/)? Or do you have any additions/suggestions for current docs state before publishing?

**Comment by aprotopopov** [comment]
Btw, do you now why there is no image for [lifetimes badge](https://badge.fury.io/for/py/lifetimes) on badge.fury.io?

**Comment by CamDavidsonPilon** [comment]
Hey @aprotopopov, thanks for being patient :)

1. Do you mind enabling changes from the maintainer on this PR? (see [docs](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/)) - then I can add to this PR as well. 

2. About the badge, not sure why. I'll investigate now.

File: README.md [hunk 1]
```diff
@@ -34,305 +35,9 @@ As emphasized by P. Fader and B. Hardie, understanding and acting on customer li
 
 Requirements are only Numpy, Scipy, Pandas, [Dill](https://github.com/uqfoundation/dill) (and optionally-but-seriously matplotlib).
```
Comment (CamDavidsonPilon) [inline README.md]: TODO: need a link to docs here.

File: docs/Quickstart.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
+## Quickstart
```
Comment (CamDavidsonPilon) [inline docs/Quickstart.md]: Think we still need this? Having docs in two places means we have a high chance of having them out of sync.

**Comment by CamDavidsonPilon** [comment]
> Moved Quickstart section from main README to separate file in docs/. What about divide Quickstart into subsections?

Yea, there likely is no easy "quickstart" anymore, but we should divide on use-cases (with/without money, discrete/continuous time). Just an idea.

**Comment by CamDavidsonPilon** [comment]
2\. that site's search is case sensitive 😕 - anyways, I've updated the badge 👍

File: docs/Quickstart.md (hunk 2 above)
Comment (aprotopopov) [inline docs/Quickstart.md]: Not duplicates (as I understand), I removed Quickstart from main README.md. See comment below.

**Comment by aprotopopov** [comment]
I reduced a lot package README.md with [that](https://github.com/CamDavidsonPilon/lifetimes/pull/114/commits/6abe57bbf8e53d72261aa29e8d7c55a8270bfed4) commit (mostly delete Quickstart section).
Also to not maintain several READMEs I changed [docs/conf.py](https://github.com/aprotopopov/lifetimes/blob/docs/docs/conf.py) with [commit](https://github.com/CamDavidsonPilon/lifetimes/pull/114/commits/dca255b112e715d6fd4a3b65ab1a8b6218774195). Now README.rst in docs/ generated automatically with `make html` and then included in `index.rst` for main docs page.  Not sure what cons could be with that structure.
One **con** is that we have two parts of the same convergence for README.md #115  and [commit](https://github.com/CamDavidsonPilon/lifetimes/pull/114/commits/dca255b112e715d6fd4a3b65ab1a8b6218774195) in docs section.

**Comment by aprotopopov** [comment]
> Yea, there likely is no easy "quickstart" anymore, but we should divide on use-cases (with/without money, discrete/continuous time). Just an idea.

So we need some plan for managing that and may be new sections. Also what about sections for each model? Something like that:
- Quickstart
- Pareto/NBD Model
- BG/NBD Model
- BG/BB Model
- MBG/NBD Model
- Gamma-Gamma Model of Monetary Value
- Plots
- API

Quickstart could be divided into subsections later. 
I like how it's done in [BTYD walkthrough](https://cran.r-project.org/web/packages/BTYD/vignettes/BTYD-walkthrough.pdf).

**Comment by aprotopopov** [comment]
Also I think in **Introduction** section there are should be small description what package do like in [BTYD](https://cran.r-project.org/web/packages/BTYD/vignettes/BTYD-walkthrough.pdf)

> The BTYD package contains models to capture non-contractual purchasing
behavior of customers—or, more simply, models that tell the story of people
buying until they die (become inactinve as customers). The main models
presented in the package are the Pareto/NBD, BG/NBD and BG/BB models,
which describe scenario of the firm not being able to observe the exact time at
which a customer drops out.

**Comment by aprotopopov** [comment]
@CamDavidsonPilon what you think about switching to NumPy docs in future? http://sphinxcontrib-napoleon.readthedocs.io/en/latest/index.html. I'd like to make all functions/classes be compatible with `sphinx-autodoc` for API. I think Numpy could be better because we have long description for each keyword, but it'll take more efforts to rewrite all docstrings...

**Comment by CamDavidsonPilon** [comment]
I like this, this is something I need to do for a few of my projects. I can get started on Friday/weekend.

**Comment by aprotopopov** [comment]
@CamDavidsonPilon what you think about merge it to docs branch for now? In next iteration we could do:
- divide Quickstart in subsections
- add API for functions  
- sections for each model 
- plots section.

**Comment by CamDavidsonPilon** [comment]
Yea let's do it. I think we can iterate faster too. 

BTW I gave you commit writes to this repo, so you can clone this repo locally and PR directly to it. (might help your workflow).

**Comment by CamDavidsonPilon** [comment]
I think we should keep the quickstart mostly as is, as that's going to satisfy 80% of use cases. We can still have sections on specific models (and submodels).

**Pull request merged**